From: DevBox UML arm64 Port
Subject: [PATCH 9c] um/kernel/process.c: arm64 elf_core_copy_task_fpregs

--- a/arch/um/kernel/process.c
+++ b/arch/um/kernel/process.c
@@ -294,8 +294,15 @@
 
 int elf_core_copy_task_fpregs(struct task_struct *t, elf_fpregset_t *fpu)
 {
+#ifdef CONFIG_X86
 	int cpu = current_thread_info()->cpu;
 
 	return save_i387_registers(userspace_pid[cpu], (unsigned long *) fpu) == 0;
+#else
+	int cpu = current_thread_info()->cpu;
+	int pid = userspace_pid[cpu];
+
+	return get_fp_registers(pid, (unsigned long *) fpu) == 0;
+#endif
 }
 
